docs/reference/modules/transport.asciidoc

```diff
@@ -102,6 +102,20 @@ and ensuring that the keepalive interval is shorter than any timeout that might
 cause idle connections to be closed, or by setting `transport.ping_schedule` if
 keepalives cannot be configured.
 
+[float]
+==== Transport Compression
+
+By default Elasticsearch network-level compression is disabled. This default
+normally makes sense for local cluster communication as compression has a
+noticeable CPU cost and local clusters tend to be setup with fast network
+connections between nodes. If compression is necessary for remote cluster
+connections only, this can be configured on a per-cluster basis using the
+`cluster.remote.${cluster_alias}.transport.compress` setting.
+
+When compression is enabled using either `transport.compress` or for a specific
+remote cluster, only the outbound requests are compressed. Elasticsearch compresses
+responses if and only-if the inbound request received was compressed.
```

Comment:
```suggestion
If compression is enabled and an inbound request is compressed, {es} also compresses the response.
```

Reply:
Not sure if it's true that the compression of responses requires compression to be enabled, but it seemed like a reasonable assumption.